src/modules/git_state.rs

```diff
@@ -11,7 +11,8 @@ use super::{Context, Module};
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let mut module = context.new_module("git_state")?;
 
-    let repo_root = context.repo_root.as_ref()?;
+    let repo = context.get_repo().ok()?;
+    let repo_root = repo.root.as_ref()?;
```

Comment:
At the moment `git_state` re-opens the repository, but perhaps you could capture `Repository.state()` on to your Repo struct...

If necessary I can do this in a subsequent PR.

Reply:
I can definitely add that.